### ISSUE TITLE
Support badges in preview

### ISSUE THREAD

**Issue by bbringhu** [AdobeDocs/adobe-markdown-authoring#5]
Badges should be supported in preview. Documentation and examples are here:

https://experienceleague.adobe.com/docs/authoring-guide-exl/using/markdown/syntax-style-guide.html?lang=en#badges

**Comment by geverit4** [comment]
Not yet implemented - examples are not available yet.

**Comment by geverit4** [comment]
I do not have access to this document.

https://experienceleague.adobe.com/docs/authoring-guide-exl/using/markdown/syntax-style-guide.html?lang=en#badges

**Comment by geverit4** [comment]
@bbringhu 

I was able to create a transformer for the inline badges, and as far as I can tell they are generating the correct HTML output, but for some reason the badge styling is not being applied.  I tried several different approaches, and since I'm using the same technique for the tabs, it *should* work.  I tried several approaches to get it to display the badge styling, but nothing worked. We  are using web components, the same as Exl.  The HTML output looks very similar to the example site.  

![Screenshot 2023-04-19 at 10 14 16 PM](https://user-images.githubusercontent.com/1187616/233240847-c6126bbd-f86d-4fd5-966b-9101cfa5e7f3.png)

Here's some example output from the new transformer: 

```Markdown
[!BADGE Beta]{type="Informative" url="https://www.example.com"}
```
renders as 
```HTML
<a href="https://www.example.com" style="color:inherit !important;text-decoration:none" title="https://www.example.com">
   <span class="sp-badge-wrapper">
      <sp-badge size="s" style="cursor:inherit !important" variant="informative" dir="ltr">Beta</sp-badge>
   </span>
</a>
```

Here's an example of the "Beta" tag element from the hidden test page:

```
<a href="https://www.example.com" style="color:inherit !important;label-decoration:none">
   <span class="sp-badge-wrapper">
      <sp-badge size="s" style="cursor:inherit !important" variant="informative" dir="ltr">Beta</sp-badge>
   </span>
</a>
```

I also attempted to follow the instructions at [Spectrum Web Components](https://opensource.adobe.com/spectrum-web-components/getting-started/) to set up a completely independent test project but no matter what I tried, the badges (and buttons for that matter) would not display with the any styling at all.

**Comment by geverit4** [comment]
Still not perfect - but they do use the styling direct from Spectrum Web Components: 

![Screenshot 2023-04-27 at 8 50 54 PM](https://user-images.githubusercontent.com/1187616/235028987-b75da889-a81f-45f4-9238-727f6260bcbf.png)

**Comment by bbringhu** [comment]
@geverit4 - I found one issue. If there's a simple badge with no parameters, it doesn't display. Example:

`[!BADGE Just Me!]`

The above doesn't work in preview, but the following does:

`[!BADGE Just Me]{url="https://www.adobe.com"}`

**Comment by geverit4** [comment]
Added support for bare badge with no parameters.

![Screenshot 2023-05-02 at 5 54 22 PM](https://user-images.githubusercontent.com/1187616/235794680-fc89020b-5c86-48d0-b2ea-26988c26e021.png)

**Comment by bbringhu** [comment]
Minor issue - Only the first inline badge in a paragraph is rendered in preview.

![badges-first-only-renders](https://github.com/AdobeDocs/adobe-markdown-authoring/assets/6193209/3e2e5c7d-c10c-4be7-bd8e-a84521d3a02a)

**Comment by geverit4** [comment]
Will be addressed in 2.0.8.
![Screenshot 2023-05-22 at 4 15 22 PM](https://github.com/AdobeDocs/adobe-markdown-authoring/assets/1187616/752521fa-248f-4e99-bd5e-423278da06d8)

**Comment by bbringhu** [comment]
Fixed in general. (any minor badge preview issue will be addressed as a separate issue)
